crates/nu-parser/src/parse.rs

```diff
@@ -1829,38 +1829,6 @@ fn parse_call(
                 error,
             );
         }
-    } else if lite_cmd.parts[0].item == "source" {
-        if lite_cmd.parts.len() != 2 {
-            return (
-                None,
-                Some(ParseError::argument_error(
-                    lite_cmd.parts[0].clone(),
-                    ArgumentError::MissingMandatoryPositional("a path for sourcing".into()),
-                )),
-            );
-        }
-        if lite_cmd.parts[1].item.starts_with('$') {
-            return (
-                None,
-                Some(ParseError::mismatch(
-                    "a filepath constant",
-                    lite_cmd.parts[1].clone(),
-                )),
-            );
-        }
-        if let Ok(contents) =
-            std::fs::read_to_string(expand_path(&lite_cmd.parts[1].item).into_owned())
-        {
-            let _ = parse(&contents, 0, scope);
```

Comment:
We still need to do this step, as it lets us see the definitions coming in from the `source` while inside the parser. Maybe we can keep this part but otherwise follow the other code paths?

Reply:
I knew it was too good to be true :grinning:. 

This seems like a chicken and egg problem. I would need to study the nushell parsing chain to understand how to fix it since I have no clue at the moment.